src/core/core.scale.js

```diff
@@ -287,8 +288,15 @@ module.exports = Element.extend({
 		helpers.callback(this.options.beforeBuildTicks, [this]);
 	},
 	buildTicks: helpers.noop,
-	afterBuildTicks: function() {
-		helpers.callback(this.options.afterBuildTicks, [this]);
+	afterBuildTicks: function(ticks) {
+		var me = this;
+		// ticks is empty for old axis implementations here
+		if (helpers.isArray(ticks) && ticks.length) {
```

Comment:
I think `if (helpers.isArray(ticks))` is enough, if `ticks` is an array, even empty, it's the new implementation?

Reply:
That's right, was again thinking external scale calling `afterBuildTicks()` directly.